src/test/scala/com/fullcontact/interview/RecordFinderTest.scala

```diff
@@ -1,9 +1,94 @@
 package com.fullcontact.interview
 
-import org.scalatest.{FunSuite, Matchers}
+import org.apache.spark.sql.SparkSession
+import org.scalatest.{BeforeAndAfter, FunSuite, Matchers}
 
-class RecordFinderTest extends FunSuite with Matchers {
-  test("todo") {
-    pending
+class RecordFinderTest extends FunSuite with Matchers with BeforeAndAfter {
+  var sparkSession: SparkSession = _
+
+  test("Validate exploding records dataframe by each identifier") {
+    val spark = sparkSession
+
+    import spark.implicits._
+
+    val records = Seq(
+      "A B C",
+      "D E F",
+      "A C D"
+    ).toDF(RecordFinder.Record)
+
+    val expected = Seq(
+      ("A", "A B C"),
+      ("B", "A B C"),
+      ("C", "A B C"),
+      ("D", "D E F"),
+      ("E", "D E F"),
+      ("F", "D E F"),
+      ("A", "A C D"),
+      ("C", "A C D"),
+      ("D", "A C D"),
+    ).toDF(RecordFinder.Identifier, RecordFinder.Record)
+
+    val exploded = RecordFinder.explodeRecordsDF(records)
+
+    assert(
+      expected.as[(String, String)].collect() sameElements exploded.as[(String, String)].collect(),
+      "The exploded dataframe was not the same as the expected value"
+    )
+
+  }
+
+  test("Validate grouping exploded dataframe by identifier") {
+    val spark = sparkSession
+    import spark.implicits._
+
+
+    val expected = Seq(
+      ("A", "A B C D"),
+      ("B", "A B C"),
+      ("C", "A B C D"),
+      ("D", "A C D E F"),
+      ("E", "D E F"),
+      ("F", "D E F")
+    ).toDS()
+      .as[(String, String)]
```

Comment:
What motivates using `.toDS` or `.toDF` when setting up your test data? 

Or, more generally, where might you prefer to use a DataFrame or specifically-typed Dataset over the other?